tests/unit/sstring_test.cc

```diff
@@ -214,3 +215,12 @@ BOOST_AUTO_TEST_CASE(test_nul_termination) {
         BOOST_REQUIRE(!strncmp(s1.c_str(), s2.c_str(), std::min(s1.size(), s2.size())));
     }
 }
+
+BOOST_AUTO_TEST_CASE(test_back_inserter) {
+    sstring key = "meaning";
+    int value = 42;
+    auto formatted_size = fmt::formatted_size("{}={}", key, value);
```

Comment:
This will likely crash and burn once the formatted output is larger than 15 characters.

Reply:
how so? if `formatted_size` is calculated by fmtlib, and `sstring` instance is preallocated using this size, i think we are safe here.

Reply:
It may work, but only if the user initializes the string correctly, and only because this isn't a real back_inserter.

std::back_inserter is something that grows the container. This back_inserter doesn't, it just overwrites the string like a regular iterator.